dashboard/app/models/concerns/pd/jot_form_backed_form.rb

```diff
@@ -107,6 +107,14 @@ def _sync_from_jotform(form_ids)
 
         form_ids.each do |form_id|
           questions = get_questions(form_id, force_sync: true)
+          questions_details = use_names_for_question_ids? ? JSON.parse(questions.questions) : nil
+          if questions_details
+            # Make sure that there is a unique, non-nil name for each question.
+            if questions_details.pluck("name").compact.uniq.size != questions_details.size
+              raise "Not all questions for form #{form_id} have unique names."
```

Comment:
I'm interested in what we can do to surface feedback like this to survey authors as early as possible.

One idea: Provide a test route (e.g. `studio.code.org/validate_jotform`) that, given a jotform id, will "lint" the form to ensure it only uses the subset of features we support, and returns red/green so form authors can check "is my form ok?" without asking an engineer or waiting for us to surface a Honeybadger error.

Reply:
+1 to prevention. Since we only detect this after there are submissions to the forms, we will have to update those submissions in JotForm if we want to ingest them.